### PR TITLE
Make BitDropDown responsive in small screens (#429)

### LINKED ISSUE #429: BitDropDown should be responsive to small screens
As the default behaviour of the counterpart component in Fluent UI framework, the BitDropDown component should be opened like the following image in response to the small screens:

### PR DESCRIPTION
**Pull request by RezaKargar**
This closes #429

### CODE REVIEW

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss [hunk 1]
```diff
@@ -289,6 +289,41 @@
     }
 }
 
+.overflow-hidden {
+    overflow: hidden !important;
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:293]: We must never use `!important` in our codes.
plz remove this keyword and if there is some issue with removing it, you should find another way to solve it. using `!important` is not the solution :)

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss [hunk 2]
```diff
@@ -289,6 +289,41 @@
...
+}
+
+@media only screen and (min-width: 0) and (max-width: rem(639px)) {
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:296]: plz, use the `em` unit while defining the media query breakpoint.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss [hunk 3]
```diff
@@ -289,6 +289,41 @@
...
+    .bit-drp-items-wrapper {
+        box-shadow: rgb(0 0 0 / 22%) 0 25.6px 57.6px 0, rgb(0 0 0 / 18%) 0 4.8px 14.4px 0;
+        position: fixed !important;
+        top: 0 !important;
+        left: unset !important;
+        right: 0 !important;
+        bottom: unset !important;
+        height: 100% !important;
+        width: rem(272px) !important;
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:305]: plz remove these `!important`s :)))

**Review by maryamhdr — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss (hunk 3 above)
Comment (RezaKargar) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:305]: Here if I want to remove the `!important` from the styles I should handle it with TS/JS but its not as efficient as css media queries,
I give you an example, consider a mobile user opens a dropdown and then rotate the phone to be in the landscape mode in this case the media query will perform correctly but in TS/JS approach the user must close and re-open the dropdown to see it in proper way

Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:305]: in these situations, we can simply close the callout. we just need to capture the appropriate events.

Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:305]: we did the same in the window resize event.

**Comment by msynk** [comment]
any updates? @RezaKargar

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss (hunk 3 above)
Comment (RezaKargar) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/DropDown/BitDropDown.Fluent.scss:305]: But in Fluent's Demos the dropdown does not close on rotate

**Comment by msynk** [comment]
I think @maryamhdr can help you here.

**Comment by maryamhdr** [comment]
> There is still a problem with my solution and that is when the user rotates the phone the dropdown closes but the `overflow: hidden` style added to **body** by `overflow-hidden` class still remains, I have handled it inside the dropdown's TS/JS script not inside the window resize handler
> 
> Any ideas? @msynk @maryamhdr

Hey Reza. I need to talk to you to clarify what is the issue you are mentioning to.
How can we talk? :)

**Comment by RezaKargar** [comment]
@Md23Mh has my contact,
I'll ask her to link us
@maryamhdr

**Comment by msynk** [comment]
any updates?
@RezaKargar @maryamhdr

File: src/Client/Web/Bit.Client.Web.BlazorUI/Scripts/bit.blazorui.ts [hunk 4]
```diff
@@ -71,7 +71,10 @@ class Bit {
 }
 
 window.addEventListener('scroll', (e: any) => {
-    if (e.target.id && Bit.currentDropDownCalloutId === e.target.id) return;
+    const minimumWidthForDropDownNormalOpen = 640;
+    if (window.innerWidth < minimumWidthForDropDownNormalOpen ||
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Scripts/bit.blazorui.ts:75]: We must checkout 2 conditions:
In the first condition, we must check if the screen size is lower than 640px and the callout which is open is the drop-down callout, then return.
In the second one, we must check if the scroll event target is the drop-down callout, then return.

**Review by maryamhdr — CHANGES_REQUESTED** [review]

**Review by maryamhdr — APPROVED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**